idx/memory/memory_test.go

```diff
@@ -91,6 +92,47 @@ func getMetricData(orgId uint32, depth, count, interval int, prefix string, tagg
 	return data
 }
 
+// getMetricDataWithCustomTags returns a count-length slice of MetricData's with random Name, the given org id, and uses custom tags
+// tags should be in the format of "key1=value1"
+func getMetricDataWithCustomTags(orgId uint32, depth, count, interval int, prefix string, tagged bool, tags []string, unique bool) []*schema.MetricData {
```

Comment:
as discussed I'ld like to see some more nuance here. There's a spectrum between "all tags identical between all metrics", and "all tags unique across all metrics".
currently this nuance is lost and very implicit. when you dig into testMemoryIndexHeapUsageWithTags one can see that even in "unique" mode, the tags are mostly identical. (because testMemoryIndexHeapUsageWithTags adds a bunch of hardcoded ones)
This seems very misleading.
I'ld like this to be clear and explicit. maybe use a uniqueness ratio between 0 and 1, and have different Test functions call with a couple ratios exploring that spectrum. And maybe only set tags in getMetricDataWithCustomTags instead of in getMetricDataWithCustomTags + its caller